### PR TITLE
Kev ma/appeals 38955

### PR DESCRIPTION
**Pull request by Kevma50287**
Resolves [38955](https://jira.devops.va.gov/browse/APPEALS-38955)

# Description
- Created `AutoAssignAlertBanner.jsx` and `BatchAutoAssignButton.jsx` components
- Added Jest tests: `AutoAssignAlertBanner.test.js` and `BatchAutoAssignButton.test.js`
- Updated ReviewPackage Redux Store to include autoAssign
- Added autoAssign reducers and actions to reviewPackage actions and reducers
- Updated `correspondence_controller#auto_assign_correspondences`
- Added `correspondence_controller#auto_assign_status`
- Added Rspec tests for auto_assign_status
- Linting and Code Climate fixes
- Added FeatureToggles to raise a banners for testing purposes
  - `auto_assign_banner_failure`
  - `auto_assign_banner_max_queue`
  - `auto_assign_banner_no_rpt`

## Acceptance Criteria
- [x] Code compiles correctly

## Testing Plan

1. Sign-in as INBOUND OPS ADMIN user
2. Go to Queue > Inbound Ops Team team cases view
3. Click on Auto assign correspondences button
4. Green Banner appears when successful
5. Click on button again
6. Blue info/error banner appears as no correspondences are available for assignment
7. In Development/Demo, go to /test/users page either through the URL or by clicking on Switch Users in the user dropdown next to Search Cases in the header
8. Add "auto-assign-banner-failure" feature toggle to the list to manually trigger a Standard Error
9. Go back to Inbound Ops Team cases view and click on banner
10. Note that a red banner appears with UUID

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Frontend
## User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

image

image

image

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [x] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [x] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [x] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [x] Are errors being caught and handled gracefully?
- [x] Are appropriate error messages being displayed to users?
- [x] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [x] Are unhandled exceptions being caught at the application level ?

### CODE REVIEW

File: Gemfile
Comment (Kevma50287) [inline Gemfile:1]: Linting fix

File: app/controllers/correspondence_controller.rb
Comment (Kevma50287) [inline app/controllers/correspondence_controller.rb:1]: Ensures only one job is being run at a time per user

File: app/jobs/auto_assign_correspondence_job.rb
Comment (Kevma50287) [inline app/jobs/auto_assign_correspondence_job.rb:1]: Refactor into single method - ensure function inputs match class parameters

File: app/models/concerns/auto_assignable.rb
Comment (Kevma50287) [inline app/models/concerns/auto_assignable.rb:1]: Linting

File: app/services/auto_assignable_user_finder.rb
Comment (Kevma50287) [inline app/services/auto_assignable_user_finder.rb:1]: Not sure if we need to manually test users that don't pass sensitivity check. This is mocked out in the Rspecs already

File: app/services/external_api/bgs_service.rb
Comment (Kevma50287) [inline app/services/external_api/bgs_service.rb:1]: Update to method, parameter was unused

File: lib/fakes/bgs_service.rb
Comment (Kevma50287) [inline lib/fakes/bgs_service.rb:1]: Updated fakes to match changes made to ExternalApi::BGSService

**Review by youfoundmanesh — APPROVED** [review]

**Pull request merged**